cinema/views.py

```diff
@@ -24,17 +28,39 @@
 class GenreViewSet(viewsets.ModelViewSet):
     queryset = Genre.objects.all()
     serializer_class = GenreSerializer
+    http_method_names = ["get", "post"]
+
+    def get_permissions(self):
+        if self.action not in ["list", "create"]:
+            raise NotFound
+        else:
+            permissions_classes = api_settings.DEFAULT_PERMISSION_CLASSES
+        return [permission() for permission in permissions_classes]
```

Comment:
Instead of defining `get_permissions` methods or `http_method_names`, inherit ViewSet classes from appropriate mixins, e.g. `ListCreateViewMixin`, and add `authentication_classes` and `permission_classes` class attributes where needed. Apply this change to all ViewSet classes